WordPress/gutenberg
# Allow dragging-and-dropping images from the inserter to image blocks

## What?


Alternative to https://github.com/WordPress/gutenberg/pull/49297.

Allow dragging-and-dropping images from the inserter to image blocks.

## Why?


Close #48919. See the issue for more info.

## How?


I added some inline review comments below for some implementation details. The main idea is derived from https://github.com/WordPress/gutenberg/pull/49297.

## Testing Instructions




1. Open a post
2. Insert a gallery block or an image block
3. Open the global inserter and navigate to the media tab
4. Select "Openverse" and drag an image from the inserter to an image block
5. Hover over an image block should show the drop placeholder, and dropping on it should replace the image with the dragged image.
6. If it's supported, it should also upload the image to the media library for same-origin access.

### Testing Instructions for Keyboard

This is a feature about drag-and-drop, so unfortunately it cannot be tested with keyboard (yet).

## Screenshots or screencast 

https://user-images.githubusercontent.com/7753001/230865064-ec99071a-635d-476b-af0c-0431e05430de.mp4
